packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -110,17 +110,29 @@ export function workerImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
     asSrc: true,
   }
 
+  const isIncludeWorkerImportMetaUrl = (code: string): boolean => {
+    if (
+      (code.includes('new Worker') || code.includes('new SharedWorker')) &&
+      code.includes('new URL') &&
+      code.includes(`import.meta.url`)
+    ) {
+      return true
+    }
+    return false
+  }
+
   return {
     name: 'vite:worker-import-meta-url',
 
+    shouldTransformCachedModule({ code }) {
+      if (isBuild && config.build.watch && isIncludeWorkerImportMetaUrl(code)) {
```

Comment:
Good catch on the ordering of the conditions. If you don't mind, can you also update https://github.com/vitejs/vite/blob/1231ec0046976843ac2f4f1ecacbbca4000769af/packages/vite/src/node/plugins/worker.ts#L221-L223

and put `config.build.watch` as the second condition?

Reply:
Glad to hear it. I've updated.